packages/ilmomasiina-frontend/src/modules/auth/actions.ts

```diff
@@ -39,13 +39,16 @@ export const login = (email: string, password: string) => async (dispatch: Dispa
   dispatch(loggingIn());
 
   try {
-    const sessionResponse = await apiFetch('authentication', {
+    const sessionResponse = await apiFetch<AdminLoginResponse>('authentication', {
       method: 'POST',
       body: {
         email,
         password,
       },
-    }) as AdminLoginResponse;
+    });
+    if (!sessionResponse) {
+      throw Error('login error:');
+    }
```

Comment:
Can this ever happen or does it always throw `ApiError`?

Reply:
This happens if the server sends a 500.

Reply:
This checks `response.ok` which should catch those as well
https://github.com/Tietokilta/ilmomasiina/blob/9c0581e83996b99263b6445eee8d45bff7051931/packages/ilmomasiina-components/src/api.ts#L62

Reply:
https://github.com/Tietokilta/ilmomasiina/blob/9c0581e83996b99263b6445eee8d45bff7051931/packages/ilmomasiina-components/src/api.ts#L65
The problem comes from the HTTP code checking done here, typescript types it out to `T | null`.

Reply:
Maybe we should then instead 
```typescript
return (response.status === 204 ? null : response.json()) as T; 
```
since the 204 -> `null` should be specified in API typings anyway when it's expected